src/Tests/Microsoft.NET.Build.Tests/GivenThatWeHaveAPackageReferenceWithAliases.cs

```diff
@@ -139,6 +139,55 @@ public void CanBuildProjectWithAPackageReferenceWithMultipleAliases()
                 .Pass();
         }
 
+        [Fact]
+        public void It_does_not_error_on_duplicate_package_names()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "ResolveAsset",
+                TargetFrameworks = "net5.0",
+                IsSdkProject = true
+            };
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var restoreCommand = new RestoreCommand(testAsset);
+            restoreCommand.Execute()
+                .Should()
+                .Pass();
+
+            string projectAssetsJsonPath = Path.Combine(
+                testAsset.Path,
+                testProject.Name,
+                "obj",
+                "project.assets.json");
+            File.WriteAllText(projectAssetsJsonPath, @"{
+  `version`: 3,
+  `targets`: {
+    `net5.0`: {
+      `Humanizer.Core/2.2.0`: {
+        `type`: `package`
+      },
+      `Humanizer.Core/2.2.1`: {
+        `type`: `package`
+      }
+    }
+  },
+  `project`: {
+    `version`: `1.0.0`,
+    `frameworks`: {
+      `net5.0`: {
+        `targetAlias`: `net5.0`
+      }
+    }
+  }
+}".Replace('`', '"'));
```

Comment:
How does an assets file with these duplicate items end up getting produced?  Is that a bug in NuGet?

I think that either this test should reproduce the conditions that cause NuGet to generate a file like this, or if we are going to hard code the contents of the assets file then the test should go in the Microsoft.NET.Build.Tasks.UnitTests project.

Reply:
Makes sense, we have had customer reports of this happening but I haven't been able to find a way to repro reliably so I've changed this to be a unit test.

Reply:
I found an issue on NuGet that mentioned how the project.assets.json is sometimes not updated. Is it possible that the duplicate package is a variation on this? This is the issue: https://github.com/NuGet/Home/issues/7041 But this could be a red herring

Reply:
This was supposed to be a repro from the VS feedback ticket: https://github.com/jogibear9988/nugetIssue

Does that repo repro the problem?

Reply:
I wasn't able to repro with that solution, it sounded like it only reproed with certain VS versions based on the discussion on the feedback ticket.

Reply:
@dsplaisted do you think it's better to figure out a reliable end to end repro or are you happy with the unit test?